Spring-Boot/Spring-Boot-Problem/src/main/java/tw/waterball/judgegirl/springboot/problem/repositories/CacheProblemRepository.java

```diff
@@ -0,0 +1,117 @@
+package tw.waterball.judgegirl.springboot.problem.repositories;
+
+import lombok.AllArgsConstructor;
+import org.springframework.data.redis.core.RedisTemplate;
+import tw.waterball.judgegirl.commons.models.files.FileResource;
+import tw.waterball.judgegirl.primitives.problem.Language;
+import tw.waterball.judgegirl.primitives.problem.LanguageEnv;
+import tw.waterball.judgegirl.primitives.problem.Problem;
+import tw.waterball.judgegirl.problem.domain.repositories.PatchProblemParams;
+import tw.waterball.judgegirl.problem.domain.repositories.ProblemQueryParams;
+import tw.waterball.judgegirl.problem.domain.repositories.ProblemRepository;
+
+import java.io.InputStream;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * @author - wally55077@gmail.com
+ */
+@AllArgsConstructor
+public class CacheProblemRepository implements ProblemRepository {
+
+    private static final String PROBLEM_KEY = "judgegirl:problems:";
```

Comment:
don't hardcode the prefix `judgegirl`
try to declare the prefix in project-wide config